trio/_path.py

```diff
@@ -182,6 +183,78 @@ async def open(self, *args, **kwargs):
         value = await trio.to_thread.run_sync(func)
         return trio.wrap_file(value)
 
+    if TYPE_CHECKING:
+        # the dunders listed in _forward_magic that aren't seen otherwise
+        __bytes__ = pathlib.Path.__bytes__
+        __truediv__ = pathlib.Path.__truediv__
+        __rtruediv__ = pathlib.Path.__rtruediv__
+
+        # Note: these are not parsed properly by jedi.
+        # It might be superior to just manually implement all the methods and get rid
+        # of all the magic wrapping stuff.
+        ArgsT = ParamSpec("ArgsT")
+        RetT = TypeVar("RetT")
+
+        # mypy complains about missing `self` argument
```

Comment:
Might want to move this to module scope and underscore-prefix it, since right now tools will also think `trio.Path.make_aync()` is something you can do. Also `ParamSpec` needs to be imported from `typing_extensions` at runtime for 3.9 and below.

Reply:
yeah good call

and thanks!